src/modules/producer.js

```diff
@@ -115,29 +144,30 @@ function checkRpc (queue, msg, options) {
 function produce(queue, msg, options) {
   //default options are persistent and durable because we do not want to miss any outgoing message
   //unless user specify it
-  options = Object.assign({ persistent: true, durable: true }, options);
-
+  options = Object.assign({persistent: true, durable: true}, options);
+  
   return this.conn.get()
-  .then((_channel) => {
-    this.channel = _channel;
-
-    //undefined can't be serialized/buffered :p
-    if (!msg) msg = null;
-
-    this.conn.config.transport.info('bmq:producer', '[' + queue + '] > ', msg);
-
-    return checkRpc.call(this, queue, parsers.out(msg, options), options);
-  })
-  .catch((err) => {
-    //add timeout between retries because we don't want to overflow the CPU
-    this.conn.config.transport.error('bmq:producer', err);
-    return utils.timeoutPromise(this.conn.config.timeout)
-    .then(() => {
-      return produce.call(this, queue, msg, options);
+    .then((_channel) => {
+      this.channel = _channel;
+      
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove the useless trailing whitespaces at the end of this line. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.dialonce.io/coding_rules#rule_key=javascript%3ATrailingWhitespace)